### PR TITLE
fixes #23644: reload wallpapers on orientation change

### LINKED ISSUE #23644: Wallpapers do not switch their version based on orientation when changed
Steps to reproduce:

1. Navigate to wallpaper settings screen
2. Rotate device

Expected Results:

Wallpapers thumbnails switch their image based on orientation

Actual Results:

Wallpaper thumbnails continue using image based on their original orientation

### PR DESCRIPTION
**Pull request by MatthewTighe**
### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt [hunk 1]
```diff
@@ -44,35 +46,47 @@ class WallpaperSettingsFragment : Fragment() {
...
         }
     }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        (view as? ComposeView)?.apply {
+            wallpaperSettings()
+        }
+    }
+
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt:89]: Nice ✨ !!

**Review by Amejia481 — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt (hunk 1 above)
Comment (MatthewTighe) [reply app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt:89]: thanks 🙂

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt [hunk 2]
```diff
@@ -44,35 +45,47 @@ class WallpaperSettingsFragment : Fragment() {
...
             }
         }
     }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        (view as? ComposeView)?.apply {
+            wallpaperSettings()
+        }
+    }
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt:87]: Have you look at doing this with `DisposableEffect(LocalConfiguration.current.orientation) {}` which is more 'Compose'-y?

**Review by gabrielluong — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt [hunk 3]
```diff
@@ -214,7 +220,10 @@ private fun WallpaperThumbnailItem(
...
     }
 
-    val bitmap = loadWallpaperResource(wallpaper)
+    var bitmap by remember { mutableStateOf(loadWallpaperResource(wallpaper)) }
+    DisposableEffect(LocalConfiguration.current.orientation) {
+        onDispose { bitmap = loadWallpaperResource(wallpaper) }
+    }
     // Completely avoid drawing the item if a bitmap cannot be loaded and is required
     if (bitmap == null && wallpaper != defaultWallpaper) return
     Surface(
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt:229]: I would add a new line in between this to visually separate out what I call the 'logical' blocks. Just makes thing easier to read when there are clear separation of code blocks (for example, block for variable init, if/when statements, Compose content).

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt [hunk 4]
```diff
@@ -214,7 +220,10 @@ private fun WallpaperThumbnailItem(
         Modifier
     }
 
-    val bitmap = loadWallpaperResource(wallpaper)
+    var bitmap by remember { mutableStateOf(loadWallpaperResource(wallpaper)) }
+    DisposableEffect(LocalConfiguration.current.orientation) {
+        onDispose { bitmap = loadWallpaperResource(wallpaper) }
+    }
```
[the remark this answers is not in the document]
Comment (MatthewTighe) [reply app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt]: Updated

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt [hunk 5]
```diff
@@ -228,9 +237,10 @@ private fun WallpaperThumbnailItem(
             .then(border)
             .clickable { onSelect(wallpaper) }
     ) {
-        if (bitmap != null) {
+        val currentBitmap = bitmap
+        if (currentBitmap != null) {
```
[the remark this answers is not in the document]
Comment (MatthewTighe) [reply app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt]: It was originally to get around the smart-casting issue with mutable variables, but I updated it to a `?.let` guard.

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt (hunk 3 above)
Comment (MatthewTighe) [reply app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt]: Updated

**Review by gabrielluong — APPROVED** [review]

**Comment by Amejia481** [comment]
@Mergifyio backport releases_v98.0.0

**Pull request merged**